test_plans/device-selector.asciidoc

```diff
@@ -0,0 +1,75 @@
+:sectnums:
+:xrefstyle: short
+
+= Test plan for device selector
+
+This is a test plan for the device selector as described in SYCL 2020 Section 4.6.1.1.
+
+Estimated development time is three days.
+
+== Testing scope
+
+=== Backend coverage
+
+All the tests described below are not backend-specific and are performed for any SYCL backend.
+
+=== Device coverage
+
+All tests construct a test device for which conformance is assessed. All the tests described below are performed once for that test device.
+
+== Tests
+
+=== Predefined selector instances
+
+The file `device_selector/device_selector_predefined.cpp` currently implements tests for the SYCL 1.2.1. predefined device selectors `default_selector`, `gpu_selector`, `accelerator_selector`, and `cpu_selector`. These are marked as deprecated and duplicated to use the SYCL 2020 predefined device selectors `default_selector_v`, `gpu_selector_v`, `accelerator_selector_v`, and `cpu_selector_v` instead.
+
+Additionally when, for CPU, GPU, and accelerator, no device is available, it is checked whether constructing a device will throw an exception with the error code `errc::runtime`.
+
+=== Aspect selector
+
+==== Aspect coverage
+
+All aspects listed in the enum class `aspect` will be tested, let's identify these aspects as `{asp1, ..., aspN}`.
+
+The CTS will cover these combinations:
+
+- All combinations where no aspect is passed in.
+- All combinations where a single aspect is used.
```

Comment:
Same confusion here.  There are no "combinations" if there is just a single aspect.